### PR TITLE
Yuqiang/home activation code

### PR DESCRIPTION
**Pull request by smallst**
an old view that currently used for home activation code only.

image

### CODE REVIEW

File: app/models/User.coco.coffee [hunk 1]
```diff
@@ -323,6 +323,21 @@ module.exports = class User extends CocoModel
...
+      return $.t('subscribe.forever') if stripe.free is true
+      return $.t('subscribe.forever') if stripe.sponsorID
+      return $.t('subscribe.forever') if stripe.subscriptionID
+      stripeEnd =  moment(stripe.free) if _.isString(stripe.free)
+      # return moment(stripe.free).utc().format('ll') if _.isString(stripe.free)
+    if products = @get('products')
+      homeProducts = @activeProducts('basic_subscription')
+      endDate = _.max(homeProducts, (p) => new Date(p.endDate)).endDate
+      productsEnd = moment(endDate)
+      return stripeEnd.utc().format('ll') if stripeEnd and stripeEnd.isAfter(productsEnd)
```
Comment (mrfinch) [inline app/models/User.coco.coffee:338]: stripeEnd is actually under if condition above and still has access here, coffee and old js feels 😢

File: app/models/User.coco.coffee [hunk 2]
```diff
@@ -323,6 +323,21 @@ module.exports = class User extends CocoModel
...
     false
 
+  premiumEndDate: ->
+    return null unless @isPremium()
+    if stripe = @get('stripe')
+      return $.t('subscribe.forever') if stripe.free is true
+      return $.t('subscribe.forever') if stripe.sponsorID
+      return $.t('subscribe.forever') if stripe.subscriptionID
+      stripeEnd =  moment(stripe.free) if _.isString(stripe.free)
+      # return moment(stripe.free).utc().format('ll') if _.isString(stripe.free)
```
Comment (mrfinch) [inline app/models/User.coco.coffee:333]: remove

File: app/templates/account/prepaid-view.pug [hunk 3]
```diff
@@ -50,42 +59,43 @@ block content
...
-                            td= code.get('maxRedeemers') - redeemed
-                            td= code.get('maxRedeemers')
-                            td(data-i18n="account.purchased")
-                          else
-                            td -
-                            td -
-                            td(data-i18n="account_prepaid.redeemed")
-                else
-                  p(data-i18n="account_prepaid.no_codes")
+      if !features.chinaHome && false
```
Comment (mrfinch) [inline app/templates/account/prepaid-view.pug:62]: assuming u forgot to remove?

**Review by mrfinch — COMMENTED** [review]
Also, test-case failed in the last commit so have a look at it!!

File: app/models/User.coco.coffee (hunk 1 above)
Comment (smallst) [reply app/models/User.coco.coffee:338]: oh yeah. let me try fix it

File: app/templates/account/prepaid-view.pug (hunk 3 above)
Comment (smallst) [reply app/templates/account/prepaid-view.pug:62]: nope. i just want to remove this part. but not sure if nick still want it. so adding a `false` to make is fake-removed :joy:

Comment (mrfinch) [reply app/templates/account/prepaid-view.pug:62]: haha ok remove it or remove false after confirming with Nick 😅

Comment (smallst) [reply app/templates/account/prepaid-view.pug]: but yeah we can remove these part and in case we still need that we can revert them from git history

**Pull request merged**